python/fledge/plugins/filter/rename/rename.py

```diff
@@ -166,14 +166,31 @@ def find_and_replace(operation, find, replace_with, ignore_case, reading):
         dict:          A processed dictionary
     """
     revised_reading_dict = {}
-    _flag = re.IGNORECASE if ignore_case == 'false' else False
-
-    def get_all_values(nested_dictionary):
-        for key, value in nested_dictionary.items():
-            if type(value) is dict:
-                get_all_values(value)
-            _datapoint = re.sub(search_pattern, replace_with, str(key), flags=_flag)
-            revised_reading_dict[_datapoint] = value
+    _ignore_case_flag = re.IGNORECASE if ignore_case == 'false' else False
+
+    def rename_reading_attributes(readings):
+        res = dict()
+        for key in readings.keys():
+            is_key_found = False
+            # Check if key is found for case insensitive option
+            if not _ignore_case_flag and re.fullmatch(search_pattern, key, re.IGNORECASE):
+                is_key_found = True
+
+            # Check if key is found for case sensitive option
+            if _ignore_case_flag and re.fullmatch(search_pattern, key):
+                is_key_found = True
+
+            if isinstance(readings[key], dict):
+                if is_key_found:
+                    res[replace_with] = rename_reading_attributes(readings[key])
+                else:
+                    res[key] = rename_reading_attributes(readings[key])
+            else:
+                if is_key_found:
+                    res[replace_with] = readings[key]
+                else:
+                    res[key] = readings[key]
+        return res
```

Comment:
I don't think we need special treatment for nested dictionary.

Reply:
I found during my testing that to keep reading structure same after modification it is required